engine/src/variable.cpp

```diff
@@ -1114,6 +1114,24 @@ MCVarref *MCVariable::newvarref(void)
 
 ////////////////////////////////////////////////////////////////////////////////
 
+static MCAutoPointer<MCNameRef[]>
+__join_paths(MCSpan<MCNameRef> p_base,
+             MCSpan<MCNameRef> p_extra)
+{
+    MCAutoPointer<MCNameRef[]> t_result =
+            new (nothrow) MCNameRef[p_base.size() + p_extra.size()];
+    if (t_result)
+    {
+        for(int i = 0; i < p_base.size(); i++)
+            t_result[i] = p_base[i];
+    
+        for(int i = 0; i < p_extra.size(); i++)
+            t_result[i + p_base.size()] = p_extra[i];
+    }
+    
+    return std::move(t_result);
```

Comment:
I gave you duff information here. `return` implicitly moves its argument, so this can just be:

```c++
return t_result;
```